Products/ZenModel/MaintenanceWindow.py

```diff
@@ -283,8 +283,15 @@ def makeInt(v, fieldName, minv=None, maxv=None, acceptBlanks=True):
             self.stopProductionState = stopProductionState
             self.skip = skip
             now = time.time()
-            if self.started and self.nextEvent(now) < now:
-                self.end()
+            if self.started:
+                if ((t + duration * 60) < now) or (t > now) or (not self.enabled):
+                    # We're running. If we should have already ended OR the start was 
+                    # moved into the future OR the MW is now disabled, end().
+                    self.end()
+            elif (t < now) and ((t + duration * 60) > now) and (self.enabled):
```

Comment:
No, the indention is correct.  It's the case when the maintenance window
hasn't started but has been rescheduled to be active now.

On Wed, Dec 2, 2015 at 3:26 PM, smousa <notifications@github.com> wrote:

> In Products/ZenModel/MaintenanceWindow.py
> <https://github.com/zenoss/zenoss-prodbin/pull/1279#discussion_r46477767>:
>
> > @@ -283,8 +283,15 @@ def makeInt(v, fieldName, minv=None, maxv=None, acceptBlanks=True):
> >              self.stopProductionState = stopProductionState
> >              self.skip = skip
> >              now = time.time()
> > -            if self.started and self.nextEvent(now) < now:
> > -                self.end()
> > +            if self.started:
> > +                if ((t + duration * 60) < now) or (t > now) or (not self.enabled):
> > +                    # We're running. If we should have already ended OR the start was
> > +                    # moved into the future OR the MW is now disabled, end().
> > +                    self.end()
> > +            elif (t < now) and ((t + duration * 60) > now) and (self.enabled):
>
> does this elif need to be indented?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zenoss/zenoss-prodbin/pull/1279/files#r46477767>.
>